x-pack/plugins/security_solution/public/detections/components/rules/schedule_item_form/index.tsx

```diff
@@ -66,12 +80,12 @@ const MyEuiSelect = styled(EuiSelect)`
   width: auto;
 `;
 
-const getNumberFromUserInput = (input: string, defaultValue = 0): number => {
+const getNumberFromUserInput = (input: string, minimumValue = 0): number => {
```

Comment:
Updating this validation so that the user can only input the `minimumValue`.

We currently have a minimum of 1 unit of time for both Interval and Lookup time when creating a rule or updating a single one, and this is the behaviour I'm replicating for updating schedules, as well.

Reply:
I'm curious, what was the previous behavior? I checked it with the old implementation and didn't notice any difference in the UI:

```ts
const getNumberFromUserInput = (input: string, minimumValue = 0): number => {
  const number = parseInt(input, 10);
  if (Number.isNaN(number)) {
    return minimumValue;
  } else {
    // return Math.max(minimumValue, Math.min(number, Number.MAX_SAFE_INTEGER));
    return Math.min(number, Number.MAX_SAFE_INTEGER);
  }
};
```

Reply:
Using the old implementation, if you pass down `1` as the minimumValue to the function, and the user types `0` into the field, then the input will be updated with `0` (since `0` is the minimum between `number` and `Number.MAX_SAFE_INTEGER` - i.e.m the actual minimumValue passed isn't respected).

Since I want to prevent users from inputting `0`, this change -and passing `1` as `minimumValue`- enforces that.

Reply:
(Also: this is for the edge case in which the user selects the inputs with the mouse and "overwrites" it with the keyboard. Doing that, the user was able to input `0` even if the minimum value for the component was set to `1`. I'm aware that even with the old implementation the user couldn't input `0` by using the up and down arrow selectors of the field.)